ext/handlers/callbackquery.go

```diff
@@ -11,27 +12,31 @@ import (
 type CallbackQuery struct {
 	AllowChannel bool
 	Filter       filters.CallbackQuery
-	Response     Response
+	Response     func(b *gotgbot.Bot, ctx *ext.Context) error
 }
 
-func NewCallback(filter filters.CallbackQuery, r Response) CallbackQuery {
-	return CallbackQuery{
+func NewCallback(filter filters.CallbackQuery, response func(b *gotgbot.Bot, ctx *ext.Context) error) *CallbackQuery {
+	return &CallbackQuery{
 		Filter:   filter,
-		Response: r,
+		Response: response,
 	}
 }
 
-// SetAllowChannel Enables channel messages for this handler.
-func (cb CallbackQuery) SetAllowChannel(allow bool) CallbackQuery {
+// SetAllowChannel enables channel messages for this handler.
+func (cb *CallbackQuery) SetAllowChannel(allow bool) *CallbackQuery {
 	cb.AllowChannel = allow
 	return cb
 }
 
-func (cb CallbackQuery) HandleUpdate(b *gotgbot.Bot, ctx *ext.Context) error {
-	return cb.Response(b, ctx)
+func (cb *CallbackQuery) HandleUpdate(b *gotgbot.Bot, ctx *ext.Context) error {
+	err := cb.Response(b, ctx)
+	if err != nil {
+		log.Printf("Error handling callback query: %v", err)
+	}
```

Comment:
Why should the callbackquery log errors when all other handlers don't? 

I'm not a fan of this pattern, I think people should add an error handler to the dispatcher, to ensure all errors are logged properly. Alternatively, they can also wrap the callbackquery handler to log just errors if necessary.

Reply:
It's indeed a matter of design preference and code organization. Logging errors within the callback query handler can clutter the handler's logic and may not be the best approach for all situations.

Reply:
This is not a matter of "preference", it is a matter of consistency and configurability.

This change introduces a new logging paradigm which is inconsistent with all other handlers. Additionally, it cannot be configured (eg, to use JSON logging over text logging, to silence certain errors but allow others, or even to assign warn/error levels). Both of these things are already handled by the dispatcher-level error handling logic.
 
Ultimately, these are large changes which should be discussed over an issue, or in the support group, before being opened in a PR without any context. I don't like having to close PRs that people have worked on without discussion, because I hate to see work go to waste - so please make sure to discuss these things first.